### PR TITLE
Tag Fixes, Tests

### PR DESCRIPTION
**Pull request by byronic**
This is a proposed fix for #678 ; I welcome any and all reviews/comments.
Adds tag fixes to match tags from StackOverflow wherever possible, as well as removing duplicate tags; adds tests to inoculate against adding certain tags.

### CODE REVIEW

File: scripts/cibuild.rb [hunk 1]
```diff
@@ -28,6 +28,55 @@ def valid_url? (url)
...
+    "docs" => "documentation",
+    "encrypt" => "encryption",
+    "fsharp" => "f#",
+    "games" => "game",
+    "golang" => "go",
+    "library" => "libraries",
+    "linters" => "linter",
+    "nodejs" => "node.js",
+    "nuget.exe" => "nuget",
+    "parser" => "parsing"
```
Comment (nicolas-raoul) [inline scripts/cibuild.rb]: Nice!

**Review by nicolas-raoul — APPROVED** [review]
Looks good to me :-)

**Comment by shiftkey** [comment]
@byronic this is excellent, thanks for digging into this! Is it possible at all to retrieve the list of labels from an API, so we can automate even more of this? @NickCraver?

**Comment by byronic** [comment]
Thanks @shiftkey ! SO does have tag retrieval in their v2.2 API. Is there an acceptable way to test incorporating those API calls -- notably the OAuth token that would be required? 

Authorization and API info:
https://api.stackexchange.com/docs/tags
https://api.stackexchange.com/docs/authentication

Some examples:
Trying a nonexistent tag: https://api.stackexchange.com/docs/tags#order=desc&sort=popular&inname=surely-nonexistent&filter=default&site=stackoverflow&run=true
A bad tag input => preferred tag result:
https://api.stackexchange.com/docs/tags#order=desc&sort=popular&inname=encrypt&filter=default&site=stackoverflow&run=true
A good tag input that has to be iterated prior to finding the right answer:
https://api.stackexchange.com/docs/tags#order=desc&sort=popular&inname=go&filter=default&site=stackoverflow&run=true

**Comment by nicolas-raoul** [comment]
Note: StackOverflow has tags for most software development technologies, but it lacks domain (non-technical) tags that describe the project itself rather than the underlying technology, such as "game", "collaboration", "sharing", "sync", etc.

**Comment by byronic** [comment]
Good point @nicolas-raoul ; perhaps best to leave it as-is then.

**Comment by byronic** [comment]
ping to @nicolas-raoul and @shiftkey -- should I resolve these merge conflicts and re-submit? Or is there anything y'all are waiting on from me?

**Comment by nicolas-raoul** [comment]
It is good for merging in my opinion.
I am not part of the organization so I can not merge.
@shiftkey : Did you have any particular reason for not merging?

**Comment by shiftkey** [comment]
> Did you have any particular reason for not merging?

Availability mostly. I think for a first pass this is fine to add better validation, and I'll figure out how to rebase this PR on top of the latest changes to address the conflicts.

**Comment by ritwik12** [comment]
@shiftkey @byronic Can we update this PR? would be good to have this

**Comment by shiftkey** [comment]
@byronic I really like where this is headed, so I rebased it on top of the current branch and went to work updating the tags to suit. I think there's plenty of chance to refine the rules, but this is a great baseline for us.

There's a preview site available at https://deploy-preview-697--up-for-grabs-test-bench.netlify.com/ which shows the changes - including the fact there are now slightly more JS projects listed than .NET:

Screen Shot 2019-09-26 at 7 32 44 PM

**Review by shiftkey — APPROVED** [review]

**Comment by shiftkey** [comment]
Gonna let this sit overnight, but given how many files were modified I plan to land this sooner rather than later...

**Comment by shiftkey** [comment]
Actually, I'm gonna expedite this because I'm too excited. Thanks again @byronic!

**Pull request merged**